### PR TITLE
Recommended Changes from Execution in AML

### PR DESCRIPTION
**Pull request by ScoGroMSFT**
- Doc update to reiterate that Q&A isn't revised when running in AML (I got burned by this)
- Added exception handling & verbose error output (which line in the jsonl file caused the error)
- Requirements.txt change (mlflow -> 2.14.1) was already committed to development so that is no longer part of this PR

Note that the GH diff of the doc update seems to have gotten confused and shows the whole doc as updated. The succinct change there (what VS Code shows in the diff) is adding the following section:

### Generating the QA data for the pipeline

The QA data is not generated as part of the AzureML pipeline, so it must be generated locally prior to
running the pipeline in AzureML. (Upon completion of generation this data is uploaded to AzureML so it
will be accessible to the pipeline when it runs in AzureML.)

It is important to re-generate QA data whenever the input data is changed. Otherwise, the previous QA data
will be used to evaluate the revised data, which is usually not what is expected.

### CODE REVIEW

**Review by ritesh-modi — APPROVED** [review]
LGTM

**Comment by julia-meshcheryakova** [comment]
Do not merge until prerelease in merged into development.

**Pull request merged**
